File: DI/DI-Dagger-Java/src/main/java/model/cafetera/Termosifon.java
package model.cafetera;

import javax.inject.Inject;
import java.util.UUID;

public class Termosifon implements Bomba {
    private final Calentador calentador;
    private final String id = UUID.randomUUID().toString();

    @Inject
    Termosifon(Calentador calentador) {
        this.calentador = calentador;
    }

    @Override
    public void bombear() {
        if (calentador.estaCaliente()) {
            System.out.println("=> => bombeando => =>");
        }
    }

    @Override
    public String toString() {
        return "Termosifon{" +
                "id='" + id + '\'' +
                ", calentador=" + calentador +
                '}';
    }
}

File: DI/DI-Dagger-Kotlin/src/main/kotlin/models/cafeteras/Termosifon.kt
package models.cafeteras

import java.util.*
import javax.inject.Inject


data class Termosifon
@Inject constructor(private val calentador: Calentador) : Bomba {
    private val id = UUID.randomUUID().toString()

    override fun bombear() {
        if (calentador.estaCaliente()) {
            println("=> => bombeando => =>")
        }
    }
}
